pkg/webhooks/regularuser/regularuser.go

```diff
@@ -225,6 +241,43 @@ func isCustomDomainAuthorized(request admissionctl.Request) bool {
 		request.Kind.Group == customDomainGroup)
 }
 
+// isNetNamespaceAuthorized check if request is authorized for NetNamespace CR
+func isNetNamespaceAuthorized(s *RegularuserWebhook, request admissionctl.Request) bool {
+	return ((utils.SliceContains("cluster-admins", request.UserInfo.Groups) ||
+		utils.SliceContains("dedicated-admins", request.UserInfo.Groups)) &&
+		request.Kind.Kind == netNamespaceKind &&
+		request.Kind.Group == netNamespaceGroup &&
+		isNetNamespaceValid(s, request))
+}
+
+// isNetNamespaceValid check if the NetNamespace is valid
+func isNetNamespaceValid(s *RegularuserWebhook, request admissionctl.Request) bool {
+	// Decode object into a NetNamespace object
+	decoder, err := admissionctl.NewDecoder(&s.s)
+	if err != nil {
+		return false
+	}
+	netNamespace := &networkv1.NetNamespace{}
+	if len(request.Object.Raw) == 0 {
+		return false
+	} else {
+		err := decoder.Decode(request, netNamespace)
+		if err != nil {
+			return false
+		}
+	}
```

Comment:
```suggestion
	}
	err = decoder.Decode(request, netNamespace)
	if err != nil {
		return false
	}
```

From linter:

> if block ends with a return statement, so drop this else and outdent its block

Reply:
This is now fixed, but you should probably fix the rest of the code base as well. For example, right [here](https://github.com/openshift/managed-cluster-validating-webhooks/blob/master/pkg/webhooks/namespace/namespace.go#L134).

Reply:
Definitely.